### ISSUE TITLE
vfs: Modification time updates are not being updated in the remote

### ISSUE THREAD

**Issue by darthShadow** [rclone/rclone#4795]
#### What is the problem you are having with rclone?

Modification time changes in the mount are not being updated to the remote.

#### What is your rclone version (output from `rclone version`)

rclone v1.54.0-beta.4901.43e8c9def.fix-drive-changes
- os/arch: linux/amd64
- go version: go1.15.5

This is from my fork at https://github.com/darthShadow/rclone/commits/fix-drive-changes which only has 1 extra commit and no changes related to this.

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

Ubuntu 18.04.5

####  Which cloud storage system are you using? (e.g. Google Drive)

Google Drive

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

```
/home/darthshadow/bin/rclone mount PersonalTD:Media "${MOUNT_DIR}" \
    --use-mmap \
    --fast-list \
    --umask=002 \
    --allow-other \
    --attr-timeout=0 \
    --buffer-size=8M \
    --async-read=false \
    --poll-interval=0 \
    --config="${CONFIG}" \
    --cache-dir="${CACHE_DIR}" \
    --dir-cache-time=96000h \
    --drive-chunk-size=256M \
    --drive-pacer-min-sleep=1ms \
    --vfs-read-chunk-size=8M \
    --vfs-read-chunk-size-limit=8M \
    --vfs-cache-mode=full \
    --vfs-write-back=1m30s \
    --vfs-cache-max-age="${TD_CACHE_AGE}" \
    --vfs-cache-max-size="${TD_CACHE_SIZE}" \
    --vfs-cache-poll-interval=5m \
    --log-level=INFO \
    --log-file="${LOG_FILE}" \
    --stats=30s \
    --stats-log-level=NOTICE \
    --rc \
    --rc-no-auth \
    --rc-addr=127.0.0.1:5573 \
    --rc-job-expire-duration=30m \
    --rc-job-expire-interval=60s \
    --tpslimit-burst=100 \
    --drive-disable-http2=true
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

[update_modtime.log](https://github.com/rclone/rclone/files/5583106/update_modtime.log)

**Comment by ncw** [comment]
What is happening is

- app opens file read/write
- app does set mod time
- rclone sets the modtime on the cache file, but not the remote file because it is open for write
- app closes the file without changing it
- rclone doesn't upload the file because the file wasn't changed so the modtime doesn't get updated

I can reproduce this using python to open the file r/w but not writing to it.

**Comment by darthShadow** [comment]
Any luck with solving this?

**Comment by ncw** [comment]
Can you have a go with this?

It is the usual 66 lines of tests to 10 lines of fix ratio when working in the VFS layer!

[v1.55.0-beta.5268.e6fa2cd10.fix-4795-vfs-set-modtime](https://beta.rclone.org/branch/fix-4795-vfs-set-modtime/v1.55.0-beta.5268.e6fa2cd10.fix-4795-vfs-set-modtime/) on branch [fix-4795-vfs-set-modtime](https://github.com/rclone/rclone/tree/fix-4795-vfs-set-modtime) (uploaded in 15-30 mins)

**Comment by ncw** [comment]
I've merged this to master now as I want it to get more testing which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.55

Would appreciate your feedback on it though :-)

**Comment by darthShadow** [comment]
Sorry about the delay, but it hasn't slipped my mind, don't worry. It's just that mount that I use for writing has been under heavy load the past few days so I couldn't get a chance to restart it.

I will be doing that tonight and then checking it over the new few days to ensure everything is working fine.
